Button: Add clarification on selections (#770)
primer/design: content/components/button.mdx

  </Dont>
</DoDontContainer>

### Selections

> Buttons are commonly used to show a choice within a `select`, [SelectPanel](/components/selectpanel) or [ActionMenu](/components/action-menu). In such instances, they are frequently paired with an internal or external label.

Reviewer: > to show a choice within a `select`

When do we use a button to show the selections from a `select`?

Author: Actually I might be wrong on this one. I can remove that.

When implementing a "loading" button state, don't remove the button from the DOM or pass the `disabled` attribute. Doing so would make it impossible to tab to the button. If the button was just focused and activated, it would reset focus. Resetting focus would disrupt the keyboard navigation flow, and creates a confusing experience for assistive technologies such as screen readers.

Once the button is activated (and is in a loading state), it get the attribute `aria-disabled="true"`.

A separate, visually hidden element should be rendered outside of the `<button>` with a message to communicate the loading status. For example, "Saving profile".

</Box>

<CustomVideoPlayer
  loop
  autoplay
  width="350"
  src="https://github.com/primer/design/assets/2313998/b138567f-3ba8-49ab-9fa1-80a4fe30b034"
/>

</Box>

This message should be in an ARIA live region, using `aria-live="polite"`. The live region must be present on page load, but the message inside the live region should only be rendered while the button is in a loading state.

> If an error prevents process from being completed, focus should be brought to an `<h2>` (or next relevant heading) of the error banner.

Reviewer: ```suggestion
If an error prevents a process from being completed, focus should be brought to the relevant heading in the error banner.
```
